lib/scrollable_bottom_sheet.dart

```diff
@@ -130,41 +132,63 @@ class ScrollableBottomSheetState extends State<ScrollableBottomSheet>
     final borderRadius =
         BorderRadius.vertical(top: Radius.circular(widget.borderRadiusTop));
 
-    return GestureListener(
-      canDrag: widget.canDrag,
-      onVerticalDragUpdate: (details) => _onDragUpdate(details),
-      onVerticalDragEnd: (details) => _onDragEnd(details),
-      onVerticalDragCancel: () => _handleDragCancel(),
-      child: MediaQuery.removePadding(
-        context: context,
-        removeTop: true,
-        child: Container(
-          decoration: BoxDecoration(
-            boxShadow: widget.shadows,
-            borderRadius: borderRadius,
-            color: widget.backgroundColor,
-          ),
-          // Use a foreground decoration to make sure we don't allocate more height.
-          foregroundDecoration: ShapeDecoration(
-            shape: NonUniformBorder(
-              topWidth: 2,
-              color: widget.borderColor,
+    return NotificationListener(
+      onNotification: (scrollNotification) {
+        if (scrollNotification is ScrollStartNotification &&
+            scrollNotification.metrics.axis == Axis.horizontal) {
+          _isHorizontalScrolling = true;
+        }
+
+        if (scrollNotification is ScrollUpdateNotification &&
+            scrollNotification.metrics.axis == Axis.horizontal) {
+          if (scrollNotification.dragDetails == null) {
+            _isHorizontalScrolling = false;
+          }
+        }
+
+        if (scrollNotification is ScrollEndNotification &&
+            scrollNotification.metrics.axis == Axis.horizontal) {
+          _isHorizontalScrolling = false;
+        }
+
+        return false;
+      },
```

Comment:
I added some docs to this section because the bottom sheet is already complex :S 
You can take the parts that you like.

```dart
    return NotificationListener<ScrollNotification>(
      // Disable vertical scrolling when the user is scrolling in a nested
      // horizontal scroll view. Otherwise we should slightly move the bottom
      // sheet vertically up and down which feels buggy.
      onNotification: (notification) {
        if (notification.metrics.axis != Axis.horizontal) {
          return false;
        }

        if (notification is ScrollStartNotification) {
          _isHorizontalScrolling = true;
        }
        if (notification is ScrollEndNotification) {
          _isHorizontalScrolling = false;
        }

        // Drag details are null if the user started a scroll and the
        // scrollview is continue to scroll without the user touching the view.
        if (notification is ScrollUpdateNotification &&
            notification.dragDetails == null) {
          _isHorizontalScrolling = false;
        }

        return false;
      },
```